### PR TITLE
Cleanup in prep for kex2

### PR DESCRIPTION
**Pull request by chrisnojima**
This is a branch off of the big desktop branch. This is the non-kex2 things i had to fix to make kex2 work and also some moving around which we really should just pull the bandaid on

- Removed all the flow things for most files ( @MarcoPolo this is what we wanted right )
- Moved tabs out of the tabs folder
- Sharing more code (store, config, etc)
- Make filenames more consistent. blah/index.js blah/index.render.mobile.js or blah.js blah.native.mobile.js

@keybase/react-hackers this and nojima/DESKTOP-thrashing should likely go asap to avoid more merge nightmares

### CODE REVIEW

File: react-native/react/actions/config.desktop.js [hunk 1]
```diff
@@ -1,26 +0,0 @@
-'use strict'
```
Comment (chrisnojima) [inline react-native/react/actions/config.desktop.js]: Share more code in config flow

File: react-native/react/actions/login.js [hunk 2]
```diff
@@ -1,15 +1,16 @@
 'use strict'
```
Comment (chrisnojima) [inline react-native/react/actions/login.js]: login2 -> login

File: react-native/react/actions/login.js [hunk 3]
```diff
@@ -91,7 +94,7 @@ function setCodePageMode (mode) {
 }
 
 function qrGenerate (code) {
-  const qr = QRCodeGen(3, 'L')
+  const qr = QRCodeGen(4, 'L')
```
Comment (chrisnojima) [inline react-native/react/actions/login.js]: we need slightly more bits

File: react-native/react/actions/login.js [hunk 4]
```diff
@@ -343,8 +345,10 @@ function startLoginFlow (dispatch, getState, provisionMethod, userPassTitle, use
       payload: error || null
     })
 
-    dispatch(navigateTo([]))
-    dispatch(switchTab(devicesTab))
+    if (!error) {
```
Comment (chrisnojima) [inline react-native/react/actions/login.js]: don't change if we error out

File: react-native/react/actions/login.js [hunk 5]
```diff
@@ -378,26 +382,32 @@ function makeKex2IncomingMap (dispatch, getState, provisionMethod, userPassTitle
...
       }))
     },
     'keybase.1.secretUi.getKeybasePassphrase': (param, response) => {
-      const { passphrase } = getState().login2.userPass
+      const { passphrase } = getState().login.userPass
       if (!passphrase) {
         dispatch(askForUserPass(userPassTitle, userPassSubtitle, () => {
-          const { passphrase } = getState().login2.userPass
-          response.result(passphrase)
+          const { passphrase } = getState().login.userPass
```
Comment (chrisnojima) [inline react-native/react/actions/login.js]: this response changed

File: react-native/react/base-react.desktop.js [hunk 6]
```diff
@@ -3,4 +3,6 @@
 export * from 'react'
 import * as React from 'react'
 export default React
-export const StyleSheet = styles => styles
+export const StyleSheet = {
```
Comment (chrisnojima) [inline react-native/react/base-react.desktop.js]: we call StyleSheet.create in RN code

File: react-native/react/base-redux.desktop.js [hunk 7]
```diff
@@ -1,5 +1,4 @@
 'use strict'
-/* @flow */
```
Comment (chrisnojima) [inline react-native/react/base-redux.desktop.js]: remove all these annotations when not really used

File: react-native/react/local-debug.desktop.js [hunk 8]
```diff
@@ -12,19 +11,22 @@ let config = {
   overrideRouterState: null,
   overrideActiveTab: null,
   skipLoginRouteToRoot: false,
-  allowStartupFailure: false
+  allowStartupFailure: false,
+  printRPC: false
```
Comment (chrisnojima) [inline react-native/react/local-debug.desktop.js]: desktop can print rpcs in console now too

File: react-native/react/nav.android.js [hunk 9]
```diff
@@ -5,13 +5,13 @@ import {connect} from './base-redux'
...
 import globalRoutes from './router/global-routes'
 
-import Folders from './tabs/folders'
-import Chat from './tabs/chat'
-import People from './tabs/people'
-import Devices from './tabs/devices'
-import NoTab from './tabs/no-tab'
-import More from './tabs/more'
-import Startup from './tabs/start-up'
+import Folders from './folders'
```
Comment (chrisnojima) [inline react-native/react/nav.android.js]: tabs got moved up

File: react-native/react/nav.desktop.js [hunk 10]
```diff
@@ -29,6 +30,12 @@ const tabs = {
 }
 
 class Nav extends Component {
+  constructor (props) {
+    super(props)
+
+    this.props.dispatch(startup())
```
Comment (chrisnojima) [inline react-native/react/nav.desktop.js]: desktop call startup flow too

File: react-native/react/reducers/config.js [hunk 11]
```diff
@@ -59,7 +58,8 @@ export default function (state: ConfigState = initialState, action: any): Config
     case LoginConstants.loginDone:
       return {
         ...state,
-        navState: Constants.navLoggedIn
+        error: action.error ? action.payload : null,
```
Comment (chrisnojima) [inline react-native/react/reducers/config.js]: don't move to loggedin state if we're errored out

File: react-native/react/store/configure-store.desktop.js [hunk 12]
```diff
@@ -1,19 +0,0 @@
-'use strict'
```
Comment (chrisnojima) [inline react-native/react/store/configure-store.desktop.js]: configure store sharing more code

File: react-native/react/styles/common.js [hunk 13]
```diff
@@ -68,7 +67,8 @@ export default StyleSheet.create({
   },
   centerText: {
     textAlign: 'center'
-  }
+  },
+  ...nativeStyles
```
Comment (chrisnojima) [inline react-native/react/styles/common.js]: allow platform to export their own styles

**Comment by cjb** [comment]
LGTM!

**Pull request merged**